source/extensions/quic/server_preferred_address/fixed_server_preferred_address_config.cc

```diff
@@ -1,13 +1,75 @@
 #include "source/extensions/quic/server_preferred_address/fixed_server_preferred_address_config.h"
 
+#include "source/common/network/utility.h"
+#include "source/common/quic/envoy_quic_utils.h"
+
 namespace Envoy {
 namespace Quic {
 
-std::pair<quic::QuicSocketAddress, quic::QuicSocketAddress>
+namespace {
+
+quic::QuicSocketAddress
+ipOrAddressToAddress(const FixedServerPreferredAddressConfig::QuicSocketOrIpAddress& address,
+                     int32_t port) {
+  return absl::visit(
+      [&](const auto& arg) -> quic::QuicSocketAddress {
+        using T = std::decay_t<decltype(arg)>;
+        if constexpr (std::is_same_v<T, quic::QuicSocketAddress>) {
+          return arg;
+        } else if constexpr (std::is_same_v<T, quic::QuicIpAddress>) {
```

Comment:
nit: Since each of these branches is returning, let's remove the "} else" which in particular removes the indentation for the last case.